wsmaster/che-core-api-factory-github/src/main/java/org/eclipse/che/api/factory/server/github/GithubFactoryParametersResolver.java

```diff
@@ -91,13 +91,15 @@ public FactoryDto createFactory(@NotNull final Map<String, String> factoryParame
       throws BadRequestException, ServerException {
 
     // no need to check null value of url parameter as accept() method has performed the check
-    final GithubUrl githubUrl = githubUrlParser.parse(factoryParameters.get(URL_PARAMETER_NAME));
+    final GithubUrl githubUrl = githubUrlParser.parse(factoryParameters.remove(URL_PARAMETER_NAME));
```

Comment:
Since `factoryParameters` is an input parameter, is it safe to mutate it this way? If it is safe, it should nevertheless be mentioned in the `FactoryParameterResolver#createFactory` javadoc.

Reply:
that is refactored now